runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -172,6 +180,7 @@ public CorfuRecordId enqueue(E e) {
          * because if this method is wrapped in a transaction, the order is established only later.
          */
         @Getter
+        @Setter
```

Comment:
You must be very careful with setters. They always bring much more complexity to the code, make it fragile, more complicated to analyze and there are always problems with concurrency.

Reply:
yes unfortunately we do not have access to the commit sequence and epoch until the transaction actually commits, so we cannot go with immutable construct. It has to be set later, creating a new copy would break it away from the SMR version.

Reply:
sad to hear that we can't afford immutability, so, ok